app/lib/analyzer/backend.dart

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:math';
+
+import 'package:gcloud/db.dart';
+
+import '../shared/utils.dart';
+
+import 'models.dart';
+
+final Random _random = new Random.secure();
+
+/// Datastore-related access methods for the analyzer service
+class AnalysisBackend {
+  final DatastoreDB db;
+
+  AnalysisBackend(this.db);
+
+  /// Gets the [Analysis] object from the Datastore. [version] and [analysis] is
+  /// optional, when they are missing the analysis from the latest rollup will
+  /// be returned.
+  Future<Analysis> getAnalysis(String package,
+      [String version, String analysis]) async {
+    final Key packageKey = db.emptyKey.append(PackageAnalysis, id: package);
+
+    Future<Analysis> loadRollup(AnalysisRollup rollup) async {
```

Comment:
It's not loading any "rollup", it's loading the latest analysis result (feeling very uncomfortable with "rollup" :-/ )